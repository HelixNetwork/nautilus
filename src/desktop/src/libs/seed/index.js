import Keychain from './keychain';

export default {
    keychain: Keychain
};