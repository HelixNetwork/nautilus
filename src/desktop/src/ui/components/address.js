import React from "react";
import classNames from "classnames";
import css from "./address.scss";
/**
 * Component to set the 
 */
export default class Address extends React.PureComponent {


render(){
  
  return  (
   <div className={(css.addressbox)}>
       <div> </div>
       
       <div>Ready</div>
   </div>
  );
};
}

