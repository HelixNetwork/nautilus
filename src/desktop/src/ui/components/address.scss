.addressbox{
    margin-left: 0vw;
    margin-top: 2vw;
    position: absolute;
    width: 100%;
    height: 13%;
    background-image: linear-gradient(to right, rgb(21, 35, 83),rgb(19, 33, 69));
}