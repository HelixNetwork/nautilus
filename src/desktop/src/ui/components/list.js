import React from "react";
import PropTypes from "prop-types";
import orderBy from "lodash/orderBy";
import map from "lodash/map";
import classNames from "classnames";
import { withI18n } from "react-i18next";
import { connect } from "react-redux";
import moment from "moment";
import { generateAlert } from "actions/alerts";
import { promoteTransaction, retryFailedTransaction } from "actions/transfers";
import { toggleEmptyTransactions } from "actions/settings";
import {
  getSelectedAccountName,
  getSelectedAccountMeta,
  getAccountNamesFromState,
  selectAccountInfo
} from "selectors/accounts";
import { getThemeFromState } from "selectors/global";
import { formatHlx, unitStringToValue } from "libs/hlx/utils";
import {
  formatTime,
  formatModalTime,
  convertUnixTimeToJSDate,
  detectedTimezone
} from "libs/date";
import SeedStore from "libs/seed";
import {
  mapNormalisedTransactions,
  formatRelevantTransactions
} from "libs/hlx/transfers";
import Clipboard from "ui/components/clipboard";
import Icon from "ui/components/icon";
import Scrollbar from "ui/components/scrollbar";
import Button from "ui/components/button";
import css from "./list.scss";
import { getAccountInfo } from "actions/accounts";
/**
 * Transaction history list component
 */

export class ListComponent extends React.PureComponent {
 
  static propTypes = {
    /** Wallet mode */
    mode: PropTypes.string.isRequired,
    /** Hide empty transactions flag */
    hideEmptyTransactions: PropTypes.bool.isRequired,
    /** Should update history */
    updateAccount: PropTypes.func,
    /** Toggle hide empty transactions */
    toggleEmptyTransactions: PropTypes.func.isRequired,

    accountInfo: PropTypes.object.isRequired,
    /** Promotes bundle
     * @param {string} bundle - bundle hash
     * @param {object} seedStore
     */
    promoteTransaction: PropTypes.func.isRequired,
    /** Retry failed bundle
     * @param {string} bundle - bundle hash
     * @param {object} seedStore
     */
    retryFailedTransaction: PropTypes.func.isRequired,
    /** Set active history item
     * @param {number} index - Current item index
     */
    setItem: PropTypes.func.isRequired,
    /** Current active history item */
    currentItem: PropTypes.string,
    /** Translation helper
     * @param {string} translationString - locale string identifier to be translated
     * @ignore
     */
    t: PropTypes.func.isRequired,
    /** @ignore */
    accountMeta: PropTypes.object.isRequired,
    /** @ignore */
    password: PropTypes.object.isRequired,
        /** @ignore */
    style: PropTypes.object,
    /** @ignore */
    ui: PropTypes.object.isRequired

  };

  state = {
    filter: "All",
    search: "",
    loaded: true,
    isBusy: false,
    isLoading: false,
    currentlyPromotingBundleHash: false,
    isRetryingFailedTransaction: false,
    transactions: []
  };

  changeFilter(e){
    this.switchFilter(e.target.value)
  }

  switchFilter(filter) {
    if (filter === this.state.filter) {
      return;
    }

    this.setState({
      filter: filter,
      loaded: false
    });
    setTimeout(() => {
      this.setState({
        loaded: true
      });
    }, 200);
  }

  listAddresses(tx) {
    const { t } = this.props;


    return (
      <div className={css.addresses}>
        <strong>{t("addresses")}:</strong>
        <Scrollbar>
          {tx.inputs.concat(tx.outputs).map((input, index) => {
            return (
              <p key={`${index}-${input.address}`}>
                <span>
                  <Clipboard
                    text={`${input.address}${input.checksum}`}
                    title={t("history:addressCopied")}
                    success={t("history:addressCopiedExplanation")}
                    address
                  >
                    {input.address}
                    <mark>{input.checksum}</mark>
                  </Clipboard>
                </span>
                <em>{formatHlx(input.value, true, true)}</em>
              </p>
            );
          })}
        </Scrollbar>
      </div>
    );
  }

  async promoteTransaction(e, bundle) {
    e.stopPropagation();

    const { accountMeta, password } = this.props;
    const seedStore = await new SeedStore[accountMeta.type](password);

    // this.props.promoteTransaction(bundle, seedStore);
    this.props.promoteTransaction(bundle, this.props.accountName, seedStore);
  }

  async retryFailedTransaction(e, bundle) {
    e.stopPropagation();

    const { accountMeta, password } = this.props;
    const seedStore = await new SeedStore[accountMeta.type](password);

    // this.props.retryFailedTransaction(bundle, seedStore);
    this.props.retryFailedTransaction(
      this.props.accountName,
      bundle,
      seedStore
    );
  }

  getAccountTransactions = accountData => {
    const addresses = map(
      accountData.addressData,
      addressData => addressData.address
    );
    const transactions = mapNormalisedTransactions(
      accountData.transactions,
      accountData.addressData
    );
    return formatRelevantTransactions(transactions, addresses);
  };

  componentDidMount() {
    const { ui, accountInfo } = this.props;
    const isBusy =
      ui.isSyncing ||
      ui.isSendingTransfer ||
      ui.isAttachingToTangle ||
      ui.isTransitioning;
    const isLoading = ui.isFetchingAccountInfo;
    const currentlyPromotingBundleHash = ui.currentlyPromotingBundleHash;
    const isRetryingFailedTransaction = ui.isRetryingFailedTransaction;
    const tx = this.getAccountTransactions(accountInfo);
    this.setState({
      isBusy: isBusy,
      isLoading: isLoading,
      currentlyPromotingBundleHash: currentlyPromotingBundleHash,
      isRetryingFailedTransaction: isRetryingFailedTransaction,
      transactions: tx
    });
  }

 
  showMessage(message){
    if(message.indexOf('{')!=-1){
      return 'Empty';
    }
    if(message.length > 15){
      return message.substring(0,15)+"...";
    }
    return message;
  }
  updateAccount = async () => {
  
    
    const { accountInfo, password, accountName, accountMeta } = this.props;
    const seedStore = await new SeedStore[accountMeta.type](
      password,
      accountName,
      accountMeta
    );
    this.props.getAccountInfo(
      seedStore,
      accountName,
      Electron.notify,
      true // Sync with quorum enabled
    );
    
    
  };

  updateTx(){
    
    const tx = this.getAccountTransactions(this.props.accountInfo);
    this.setState({
      transactions:tx
    })
  }

  componentDidUpdate(){
    const {transactions} = this.state;
    const tx = this.getAccountTransactions(this.props.accountInfo);
    if(tx.length>transactions.length){
    this.setState({
      transactions:tx
    })
  }
  }


  render() {
    const {
      mode,
      hideEmptyTransactions,
      toggleEmptyTransactions,
      setItem,
      currentItem,
      t,
      style
    } = this.props;

    const {
      filter,
      isBusy,
      isLoading,
      loaded,
      currentlyPromotingBundleHash,
      isRetryingFailedTransaction,
      transactions,
      search
    } = this.state;
    const filters = ["All", "Sent", "Received", "Pending"];

    const totals = {
      All: 0,
      Sent: 0,
      Received: 0,
      Pending: 0
    };

    const filteredTransactions = orderBy(transactions, "timestamp", [
      "desc"
    ]).filter(transaction => {
      const isReceived = transaction.incoming;
      const isConfirmed = transaction.persistence;

      if (hideEmptyTransactions && transaction.transferValue === 0) {
        return false;
      }

      if (
        search.length &&
        transaction.message.toLowerCase().indexOf(search.toLowerCase()) < 0 &&
        transaction.bundle.toLowerCase().indexOf(search.toLowerCase()) !== 0 &&
        !(
          search[0] === ">" &&
          unitStringToValue(search.substr(1)) < transaction.transferValue
        ) &&
        !(
          search[0] === "<" &&
          unitStringToValue(search.substr(1)) > transaction.transferValue
        ) &&
        transaction.transferValue !== unitStringToValue(search)
      ) {
        return false;
      }

      totals.All++;

      if (!isConfirmed) {
        totals.Pending++;
        if (filter === "Pending") {
          return true;
        }
      } else if (isReceived) {
        totals.Received++;
        if (filter === "Received") {
          return true;
        }
      } else {
        totals.Sent++;
        if (filter === "Sent") {
          return true;
        }
      }

      return filter === "All";
    });

    const activeTx = currentItem
      ? filteredTransactions.filter(tx => tx.bundle === currentItem)[0]
      : null;
    const isActiveFailed = activeTx && activeTx.broadcasted === false;

    const scrollStyle = {
      paddingRight: '0px',
      marginLeft: '-50px',
      width: '109%',
    };
    

    return (
      <React.Fragment>
        <nav className={css.nav}>
       
        <div className={css.search}>
          <div
              onClick={() => this.setState({ search: "" })}
              style={{ display: "inline-block", marginLeft: "-131%",color:'white',opacity:'0.3', marginTop:'5px' }}
            >
              <Icon
                icon={search.length > 0 ? "cross" : "search"}
                size={search.length > 0 ? 16 : 20}
              />
            </div>
            <input
              className={css.search_text}
              value={search}
              placeholder="Type text here..."
              onChange={e => this.setState({ search: e.target.value })}
            />
            
          </div>
          {/* Should be changed to isLoading and isBusy */}
          <a
            onClick={()=>{this.updateAccount();this.updateTx()}}
            className={classNames(css.refresh, (this.props.ui.isSyncing || this.props.ui.isSendingTransfer || this.props.ui.isAttachingToTangle || this.props.ui.isTransitioning) ? css.busy : null, this.props.ui.isFetchingAccountInfo ? css.loading : null)}
          >
            <Icon icon="sync" size={24} />
          </a>
          <p className={css.sort_by}>Sort By</p>
          <div className={css.search}><select className={css.sort_text} onChange={this.changeFilter.bind(this)}>
                  {filters.map(item => {
                  return (
                  <option value={item} key={item}>
                    {item}
                  </option>)
                })}
                </select>
                </div> 

          
        </nav>
      

          <Scrollbar style={scrollStyle}>
            {filteredTransactions.length ? (
              filteredTransactions.map((transaction, key) => {
                const isReceived = transaction.incoming;
                const isConfirmed = transaction.persistence;

                return (
                  <a
                    key={key}
                    onClick={() => setItem(transaction.bundle)}
                  >
                    {isConfirmed ?(
                    <div className={isReceived?css.column_receive:css.column_sent}>
                    <div className={css.column_cnt}>
                        <h4 className={css.sent_heading}>{isReceived ? 'RECEIVED': 'SENT'}</h4>
                        <h6>{moment.unix(transaction.timestamp).format("DD MMM YYYY")}</h6>
                       
                    </div>
                    <div className={css.column_cnt}>
                        <p className={css.note}>{this.showMessage(transaction.message)}</p>
                    </div>
                    <div className={css.column_cnt}>
                        <h4 className={css.sender_heading}>Hash</h4>
                        <p className={css.fromhash}>{transaction.bundle}</p>
                    </div>
                    <div className={css.column_cnt}>
                        <span className={isReceived?css.receive:css.sent}>{transaction.transferValue === 0
                          ? ""
                          : isReceived
                          ? "+"
                          : "-"}
                        {formatHlx(transaction.transferValue, true, true)}</span>
                    </div>
                    </div>
                    ): (
                      <div className={css.column_pending}>
                    <div className={css.column_cnt}>
                        <h4 className={css.sent_heading}>PENDING</h4>
                        <h6>{moment.unix(transaction.timestamp).format("DD MMM YYYY")}</h6>
                        
                    </div>
                    <div className={css.column_cnt}>
                        <p className={css.note}>{this.showMessage(transaction.message)}</p>
                    </div>
                    <div className={css.column_cnt}>
                        <h4 className={css.sender_heading}>Hash</h4>
                        <p className={css.fromhash}>{transaction.bundle}</p>
                    </div>
                    <div className={css.column_cnt}>
                        <span className={css.pending}>{transaction.transferValue === 0
                          ? ""
                          : isReceived
                          ? "+"
                          : "-"}
                        {formatHlx(transaction.transferValue, true, true)}</span>
                    </div>
                    </div>
                    )}

                 
                  </a>
                );
              })
            ) : (
              <p className={css.empty}>
                {!transactions.length
                  ? t("noTransactions")
                  : t("history:noTransactionsFound")}
              </p>
            )}
          </Scrollbar>
      
        <div
          className={classNames(css.popup, activeTx ? css.on : null)}
          onClick={() => setItem(null)}
        >
          <div>
            {activeTx ? (
              <div
                className={css.sent}
              >
                <p>
                  <strong>
                    {activeTx.incoming
                      ? t("history:receive")
                      : t("history:send")}{" "}
                    <span>
                      {formatHlx(activeTx.transferValue, false, true)}
                    </span>
                  </strong>
                  <small>
                    {!activeTx.persistence
                      ? t("pending")
                      : activeTx.incoming
                      ? t("received")
                      : t("sent")}
                    <em>
                      {formatModalTime(
                        navigator.language,
                        detectedTimezone,
                        convertUnixTimeToJSDate(activeTx.timestamp)
                      )}
                    </em>
                  </small>
                </p>
                <h6>{t("bundleHash")}:</h6>
                <p className={css.hash}>
                  <Clipboard
                    text={activeTx.bundle}
                    title={t("history:bundleHashCopied")}
                    success={t("history:bundleHashCopiedExplanation")}
                  />
                </p>
                {mode === "Advanced" && this.listAddresses(activeTx)}
                <div className={css.message}>
                  <strong>{t("send:message")}:</strong>
                  <Scrollbar>
                    <Clipboard
                      text={activeTx.message}
                      title={t("history:messageCopied")}
                      success={t("history:messageCopiedExplanation")}
                    />
                  </Scrollbar>
                </div>
                {!activeTx.persistence && (
                  <nav>
                    {isActiveFailed && (
                      <Button
                        className="small"
                        loading={isRetryingFailedTransaction}
                        onClick={e =>
                          this.retryFailedTransaction(e, activeTx.bundle)
                        }
                      >
                        {t("retry")}
                      </Button>
                    )}
                    {!isActiveFailed && (
                      <Button
                        className="small"
                        loading={
                          currentlyPromotingBundleHash === activeTx.bundle
                        }
                        onClick={e =>
                          this.promoteTransaction(e, activeTx.bundle)
                        }
                      >
                        {t("retry")}
                      </Button>
                    )}
                  </nav>
                )}
              </div>
            ) : null}
          </div>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  seedIndex: state.wallet.seedIndex,
  accounts: state.accounts,
  accountName: getSelectedAccountName(state),
  theme: getThemeFromState(state),
  accountMeta: getSelectedAccountMeta(state),
  accountNames: getAccountNamesFromState(state),
  accountInfo: selectAccountInfo(state),
  mode: state.settings.mode,
  ui: state.ui,
  hideEmptyTransactions: state.settings.hideEmptyTransactions,
  password: state.wallet.password,
});

const mapDispatchToProps = {
  toggleEmptyTransactions,
  promoteTransaction,
  retryFailedTransaction,
  generateAlert,
  getAccountInfo
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withI18n()(ListComponent));