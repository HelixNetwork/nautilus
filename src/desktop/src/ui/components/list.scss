.nav {
    display: flex;
    margin: 0px 20px;
    height: 10%;
    text-align: center;
    color: black;
    overflow: hidden;
    margin-top: 60px;
    margin-left: -184px;
    > a {
        padding-top: 1vw;
        margin-left: 3vw;
        color: var(--body-bg);
        margin: 0 22px 0 0;
        font-weight: 400;
        cursor: pointer;
        &.empty {
            color: var(--body-alt);
        }
        &.disabled {
            color: var(--body-alt);
            pointer-events: none;
        }
        small {
            font-size: 13px;
            margin: 0 2px;
        }
    }
    > ul > li > a {
        cursor: pointer;
    }
}

.list {
    margin-top: 0px;
    width: 65%;
    color: black;
    border-top: 4px solid #eaac32;
    margin-left: 17%;
    a {
        list-style: none;
        font-size: 14px;

        > div {
            list-style: none;
            display: flex;
            line-height: 32px;
            cursor: pointer;
            width: 100%;
            padding: 0 10px;
            > span {
                width: 50%;
                padding: 0 2%;
            }
            > span:nth-child(1) {
                width: 30px;
                margin-top: 10px;
                padding-left: 0px;
                @media (max-width: 320px) {
                    padding-left: 10px;
                }
            }
            > span:nth-child(2) {
                text-align: left;
                width: 50%;
            }
            > strong:nth-child(3) {
                width: 20%;
                text-align: left;
            }
            > span:nth-child(4) {
                width: 25%;
                text-align: right;
            }
        }
        
        &:hover > div {
            background: var(--secondary);
        }
        &.received {
            span:nth-child(1),
            span:nth-child(4) {
                color: var(--input-bg);
            }
        }
        &.sent {
            span:nth-child(1),
            span:nth-child(4) {
                color: gray;
            }
        }
        &.pending {
            div {
                opacity: 0.5;
            }
        }
    }
}

.refresh {
    display: block;
    text-align: center;

    &:hover {
        color: var(--input-bg);
    }

    &.busy {
        pointer-events: none;
        color: var(--body-alt);
    }

    &.loading {
        pointer-events: none;
        animation: spin 0.8s ease infinite;
    }
}



.popup {
    position: absolute;
    top: 0px;
    left: 0px;
    width: 100%;
    height: 100%;
    background: var(--box-body);
    opacity: 0;
    pointer-events: none;
    transition: opacity 0.15s ease-out;
    cursor: pointer;
    z-index: 3;
    > div {
        position: absolute;
        top: 50%;
        left: 14%;
        width: calc(100% - 405px);
        height: 80%;
        transform: translate(0, -50%) scale(0.9);
        background: var(--body-bg);
        padding: 20px;
        border-radius: 4px;
        transition: transform 0.2s cubic-bezier(0, 0, 0.2, 1);
        p {
            width:70%;
            &:first-child {
                display: flex;
                justify-content: space-between;
            }

            strong {
                padding-top: 21px;
            }

            small {
                font-size: 12px;
                text-align: right;
            }

            em {
                display: block;
                font-style: normal;
                color: var(--body);
            }

            &.hash {
                margin-top: 20px;
                display: block;
                font-family: 'SourceCodePro';
                word-break: break-word;
                > span {
                    display: block;
                }
            }
        }

        .message {
            display: block;
            margin: 26px 90px 20px;
            word-break: break-word;
            height: 70px;
            position: relative;
            strong {
                display: block;
                margin-bottom: 4px;
                font-size: 12px;
                padding-top: 0px;
            }
        }

        h6 {
            font-size: 12px;
            margin-bottom: 0px;
            padding-bottom: 7px;
            margin-left: 87px;
            color:white;
        }

        .received {
            p:first-child strong {
                color: var(--primary);
            }
        }
        .sent {
            p:first-child strong {
                color: var(--secondary);
            }
        }
        .pending {
            p:first-child small {
                color: var(--negative);
            }
        }
        .confirmed {
            p:first-child small {
                color: var(--positive);
            }
        }
    }
    &.on {
        opacity: 1;
        pointer-events: all;
        background: #101c37;
        > div {
            transform: translate(0, -50%) scale(1);
        }
    }

    .addresses {
        strong {
            display: block;
            margin-bottom: 8px;
            font-size: 12px;
        }
        p {
            display: flex;
            justify-content: space-between;
            font-size: 12px;
            line-height: 14px;
            font-family: 'SourceCodePro';
            word-break: break-word;

            em {
                flex: 0 0 20%;
                text-align: right;
                font-family: 'SourceSansPro';
                font-weight: 400;
            }
            mark {
                color: var(--primary);
            }
        }
        > div {
            max-height: calc(100vh - 640px);
            width: calc(100% + 20px);
            position: relative;
            overflow: hidden;

            @media (max-width: 940px) {
                max-height: calc(100vh - 680px);
            }
        }
    }

    nav {
        text-align: center;
        button:first-of-type {
            margin-bottom: 10px;
            min-width: 120px;
        }

        @media (max-width: 320px) {
            display: none;
        }
    }
}

.search{
    width: 45%;
    padding-left: 0px;
    display: inline-block;
    margin-bottom: -21px;
    margin-top: 8px;
  }
  
  .search_text{
    float: left;
    padding-left: 75px;
    font-family: Yu Gothic;
    opacity: 2.3;
    font-size: 17px;
    border: none;
    background: transparent;
    margin-left: 155px;
    color:white;
  }
  
  
  .sort_text{
    float: right;
    font-family: Yu Gothic;
    opacity: 0.3;
    font-size: 12px;
    margin-top: 10px;
    margin-bottom: 2vw;
    background: black;
    border: none;
    color: white;
    margin-right: -100px;
  }
  .sort_by{
    right: 150px;
    width: 66px;
    position: absolute;
    margin-top: 13px;
    color: white;
    opacity: 0.3;
    font-size: 14px;
  }
  .column_sent {
    float: left;
    width: 90%;
    padding: 10px;
    height: 70px;
    margin: 10px 40px 8px 40px;
    background-image: linear-gradient(to right, #661f47, #311846); 
  }
  
  
  
  .column_sent h6{
    margin-top: 0px;
    padding-left: 9px;
    font-size: 12px;
    font-family: Helvetica-Bold;
    opacity: 0.8;
    margin-left: 0px;
  }
  
  .receiver{
    margin-top: 14px;
    padding-left: 9px;
    font-size: 12px;
    letter-spacing: 4px;
    font-family: Yu Gothic;
    font-weight: unset;
  }
  
  .from{
    margin-top: -22px;
    padding-left: 9px;
    font-size: 10px;
    font-family: Helvetica-Bold;
    opacity: 0.3;
    margin-left: 0px;
  }
  
  .fromhash{
    margin-top: -28px;
    padding-left: 0;
    font-size: 12px;
    font-family: Helvetica-Bold;
    opacity: 0.6;
    margin-left: -50px;
  }
  
  .note{
    padding-left: 9px;
    font-size: 10px;
    font-family: Helvetica-Bold;
    opacity: 0.6;
    text-align: center;
    position: absolute;
    margin-top: -22px;
    margin-left: 0px;
  }
  
  .column_receive {
    float: left;
    width: 90%;
    padding: 10px;
    height: 70px;
    margin: 10px 40px 8px 40px;
    background-image: linear-gradient(to right, #152353, #132145); 
  }
  
  .column_receive h6{
    margin-top: 0px;
    padding-left: 9px;
    font-size: 12px;
    font-family: Helvetica-Bold;
    opacity: 0.8;
    margin-left: 0px;
  }

  .column_pending {
    float: left;
    width: 90%;
    padding: 10px;
    height: 70px;
    margin: 10px 40px 8px 40px;
    background-image: linear-gradient(to right, #b18907cb, #7e7c049c); 
  }
  
  .column_pending h6{
    margin-top: 0px;
    padding-left: 9px;
    font-size: 12px;
    font-family: Helvetica-Bold;
    opacity: 0.8;
    margin-left: 0px;
  }
  
  .column_cnt{
      width: 23%;
      display: inline-block;
      margin-top: 10px;
  }
  
  .row:after {
    content: "";
    display: table;
    clear: both;
  }
  
  .sent{
    background: #261542;
    float: right;
    margin-bottom: 0px;
    font-size: 15px;
    padding: 9px;
    margin-top: 11px;
    border-radius: 5px;
  }
  
  .sent_heading{
    margin-top: 0px;
    margin-bottom: 0px;
    padding-left: 10px;
    font-family: Yu Gothic;
    opacity: 0.8;
    font-size: 17px;
    font-style: unset;
    letter-spacing: 2px;
  }
  
  .sender_heading{
    margin-top: -50px;
    margin-bottom: 0px;
    padding-left: 95px;
    font-family: Yu Gothic;
    opacity: 0.8;
    font-size: 17px;
    font-style: unset;
    letter-spacing: 2px;
    text-align: center;
    position:absolute;
  }
  
  .receive{
      background: #1b2b6a;
      float: right;
        margin-bottom: 0px;
        font-size: 15px;
        padding: 9px;
        margin-top: 11px;
        border-radius: 5px;
  }
  
  .pending{
    background: #ad9a1d;
    float: right;
      margin-bottom: 0px;
      font-size: 15px;
      padding: 9px;
      margin-top: 11px;
      border-radius: 5px;
}