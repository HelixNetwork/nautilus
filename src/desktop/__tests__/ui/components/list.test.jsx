import React from 'react';
import { shallow } from 'enzyme';

import {ListComponent}  from 'ui/components/list';

import * as dateLib from 'libs/date';
dateLib.formatTime = () => 'DD/MM/YYYY';

const props = {
    isBusy: false,
    mode: 'Standard',
    isLoading: false,
    currentlyPromotingBundleHash: '',
    isRetryingFailedTransaction: false,
    hideEmptyTransactions: false,
    updateAccount: jest.fn(),
    toggleEmptyTransactions: jest.fn(),
    accountInfo : {},
    promoteTransaction: jest.fn(),
    retryFailedTransaction: jest.fn(),
    setItem: jest.fn(),
    t: (str) => str,
    accountMeta: {
        type: 'keychain',
    },
    password: {},
    ui:{
    }
};

describe('List component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<ListComponent {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    // test('List all history items', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     expect(wrapper.find('.list a')).toHaveLength(6);
    // });

    // test('Filter received items', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     wrapper.setState({ filter: 'Received' });
    //     expect(wrapper.find('.list a')).toHaveLength(1);
    // });

    // test('Filter sent items', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     wrapper.setState({ filter: 'Sent' });
    //     expect(wrapper.find('.list a')).toHaveLength(2);
    // });

    // test('Filter pending items', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     wrapper.setState({ filter: 'Pending' });
    //     expect(wrapper.find('.list a')).toHaveLength(3);
    // });

    // test('Filter items by value search', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     wrapper.setState({ search: '500000' });
    //     expect(wrapper.find('.list a')).toHaveLength(2);
    // });

    // test('Filter items by message search', () => {
    //     const wrapper = shallow(<ListComponent {...props} />);

    //     wrapper.setState({ search: 'dolor' });
    //     expect(wrapper.find('.list a')).toHaveLength(1);
    // });

    // test('Display single transaction', () => {
    //     const mockProps = Object.assign({}, props, {
    //         currentItem: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
    //     });
    //     const wrapper = shallow(<ListComponent {...mockProps} />);

    //     expect(wrapper.find('.popup').hasClass('on')).toBeTruthy();
    // });
});
