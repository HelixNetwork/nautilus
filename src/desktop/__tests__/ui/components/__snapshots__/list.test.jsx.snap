// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for List component Render the List component 1`] = `
<Fragment>
  <nav
    className="nav"
  >
    <div
      className="search"
    >
      <div
        className="search_icon_style"
        onClick={[Function]}
      >
        <Icon
          icon="search"
          size={20}
        />
      </div>
      <input
        className="search_text"
        onChange={[Function]}
        placeholder="Type text here..."
        value=""
      />
    </div>
    <div
      className="nav_div"
    >
      <p
        className="title_refresh"
      >
        Click To Refresh:
      </p>
      <div
        className="refresh"
        onClick={[Function]}
      >
        <Icon
          icon="sync"
          size={24}
        />
      </div>
    </div>
    <div
      className="nav_div"
    >
      <p
        className="title_sort"
      >
        Sort By:
      </p>
      <div
        className="sort_by"
      >
        <select
          className="sort_text"
          onChange={[Function]}
        >
          <option
            key="All"
            value="All"
          >
            All
          </option>
          <option
            key="Sent"
            value="Sent"
          >
            Sent
          </option>
          <option
            key="Received"
            value="Received"
          >
            Received
          </option>
          <option
            key="Pending"
            value="Pending"
          >
            Pending
          </option>
        </select>
      </div>
    </div>
  </nav>
  <Scrollbar
    style={
      Object {
        "marginLeft": "-50px",
        "paddingRight": "0px",
        "width": "109%",
      }
    }
  >
    <div
      className="lists"
      key="0"
      onClick={[Function]}
    >
      <div
        className="column_sent"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            SENT
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Lorem ipsum
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            0a2955eb6f6a240336b7a53b9b89cf4f588ed2847048b150c5622f89540f0675
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="sent"
          >
            0 HLX
          </span>
        </div>
      </div>
    </div>
    <div
      className="lists"
      key="1"
      onClick={[Function]}
    >
      <div
        className="column_pending"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            PENDING
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Dolor sit amet
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            b2b43b3b04c06d640cb0b418817b713f11d6b5080e122f42c80efae73f44adda
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="pending"
          >
            +
            500 kHLX
          </span>
        </div>
      </div>
    </div>
    <div
      className="lists"
      key="2"
      onClick={[Function]}
    >
      <div
        className="column_receive"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            RECEIVED
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Lorem ipsum
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            078960aba0d1379a0bdc7df25e974e156ca016b5b16d3c77e1ca695ce7b3b835
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="receive"
          >
            +
            500 kHLX
          </span>
        </div>
      </div>
    </div>
    <div
      className="lists"
      key="3"
      onClick={[Function]}
    >
      <div
        className="column_sent"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            SENT
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Lorem ipsum
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            2028ab80df9a63e7d41c39219b720a7be4c0b037f74f44c2d36f6ce01eaf5117
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="sent"
          >
            0 HLX
          </span>
        </div>
      </div>
    </div>
    <div
      className="lists"
      key="4"
      onClick={[Function]}
    >
      <div
        className="column_pending"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            PENDING
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Lorem ipsum
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            fc95e8ce0ad3bf66b0af5de18c4b7d5c3f5a286221349b102fd51513670a3dbd
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="pending"
          >
            -
            100 kHLX
          </span>
        </div>
      </div>
    </div>
    <div
      className="lists"
      key="5"
      onClick={[Function]}
    >
      <div
        className="column_pending"
      >
        <div
          className="column_cnt"
        >
          <h4
            className="sent_heading"
          >
            PENDING
          </h4>
          <h6>
            08 Jan 2019
          </h6>
        </div>
        <div
          className="column_cnt"
        >
          <p
            className="note"
          >
            Lorem ipsum
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <h4
            className="sender_heading"
          >
            Hash
          </h4>
          <p
            className="fromhash"
          >
            38accd133327f0cdd345b6a657fd48082979cf585088fa87e11d61ea9973f759
          </p>
        </div>
        <div
          className="column_cnt"
        >
          <span
            className="pending"
          >
            -
            100 kHLX
          </span>
        </div>
      </div>
    </div>
  </Scrollbar>
  <div
    className="popup"
    onClick={[Function]}
  >
    <div />
  </div>
</Fragment>
`;
